### ISSUE TITLE
implement arbitrary priors (1D)

### ISSUE THREAD

**Issue by paulthebaker** [nanograv/enterprise#148]
We can use `scipy.stats.rv_histogram()` to implement arbitrary priors.  `Prior` can already take an arbitrary RV instance, but `Parameter` is set up to use class factories.

`scipy.stats.rv_histogram` returns and RV instance with `pdf`, `cdf`, `sample`, etc methods.  In addition to working with explicit histograms, it can take in an arbitrary function sampled at discrete points to use as the PDF.

```python
hist = np.histogram(samples)
my_RV = scipy.stats.rv_histogram(hist)
```

Note that `numpy.histogram` returns (weights, bins), with the bins defining both endpoints so for `n` weights there are `n+1` bins.   An RV can be generated from an arbitrary function in this way:

```python
xs = np.linspace(left, right, 1000)
data = my_func(xs[:-1])  # generate 1 fewer point...
my_RV = scipy.stats.rv_histogram((data, xs))
```

```python
my_prior = Prior(my_RV)
```

To generate an `enterprise.Parameter` we need a class factory.  The obvious use is for empirical priors, so in a new branch on my fork I've done this as a class called `Empirical`.

**Comment by jellis18** [comment]
This sounds good but we will want it to be compatible with #119, whenever that is ready. I think your scheme would fit perfectly into the `UserParameter` in that PR.

**Comment by paulthebaker** [comment]
It think it will fit into `UserParameter` too.  What's the status of that PR?  It's getting out of date with the base branch.  Is it time to cherry-pick some of those commits that are working and start over?

**Comment by jellis18** [comment]
Yes it is getting out of date. The status of the PR is basically we need a user guide as to what it can do and how to use it (I've mostly forgotten at this point) and it needs tests.
